### ISSUE TITLE
Globby finds files in lerna packages node_modules

### ISSUE THREAD

**Issue by Raigen** [babel/babel-upgrade#27]
In a lerna monorepo the root's `node_modules` will be ignored but not `node_modules` from the packages. Is this intended behavior?

I am using my babel6 test repo https://github.com/Raigen/babel6-tmp
```js
globby(['**/.babelrc', '!./node_modules/**'])

[ 'packages/pck1/.babelrc',
  'packages/pck1/node_modules/pck2/.babelrc',
  'packages/pck2/.babelrc' ]
```
With this pattern it works as I would expect it.
```js
globby(['**/.babelrc', '!**/node_modules/**'])

[ 'packages/pck1/.babelrc', 'packages/pck2/.babelrc' ]
```

Especially for the `package.json` this becomes a long list when dependencies are already installed via `lerna bootstrap` when I run `npx babel-upgrade` in the repo.
This does also caused me problems in our main project. I had a `dist` directory with copied `node_modules` still lying around and that was why `babel-upgrade` did not updated the `.babelrc`, because it found a lot of `package.json` and it does not update `.babelrc` when more than one `package.json` is found.

**Comment by hzoo** [comment]
I didn't think about supporting monorepos yet, so anything related to that isn't intended to work yet! Thanks for a report though. I guess that's the fun thing about these. Yeah there's a checkbox in the readme " monorepo support". If you'd like to work on that go ahead though!

**Comment by Raigen** [comment]
I already planned to do a PR for it, just wanted to doublecheck that this is not intended and my fix does not breaks other usages.

This can also occure in non-monorepos. We used to have a git-ignored `dist` directory which holds a copy of `node_modules`, so it also found those `package.json` files and decided to not touch the `.babelrc`.

While testing this out with our different projects I also found one project that puts the `package.json` into the `build` directory. I am not sure how to ignore this in all cases. Looking up `.gitignore`?
At least as short term solution a warning for developers to cleanup their project folder before running `babel-upgrade` might be a good idea.
